format-delimited/src/main/java/io/cdap/plugin/format/delimited/input/PathTrackingDelimitedInputFormat.java

```diff
@@ -69,7 +69,48 @@ public boolean nextKeyValue() throws IOException, InterruptedException {
         if (delegate.nextKeyValue()) {
           // skip to next if the current record is header
           if (skipHeader && delegate.getCurrentKey().get() == 0L) {
-            return delegate.nextKeyValue();
+            delegate.nextKeyValue();
+          }
+          // getCurrentValue logic moved to nextKeyValue to avoid discrepancies in the data if there are multiple
+          // calls to getCurrentValue method.
+          String delimitedString = delegate.getCurrentValue().toString();
+          builder = StructuredRecord.builder(schema);
+          Iterator<Schema.Field> fields = schema.getFields().iterator();
+          Iterator<String> splitsIterator = getSplitsIterator(enableQuotesValue, delimitedString, delimiter);
+
+          while (splitsIterator.hasNext()) {
+            String part = splitsIterator.next();
+            if (!fields.hasNext()) {
+              int numDataFields = 0;
+              splitsIterator = getSplitsIterator(enableQuotesValue, delimitedString, delimiter);
```

Comment:
this is no longer safe to do with multi-line, as we could have advanced the delegate already. It doesn't look like this was required to create this in the first place, we could have just kept a counter instead of recreating the iterator.

Reply:
used a counter here.